src/ore/src/iter.rs

```diff
@@ -41,10 +41,36 @@ where
             Some(v1) => self.all(|v2| v1 == v2),
         }
     }
+
+    /// Converts the the iterator into an `ExactSizeIterator` reporting the given size.
+    fn exact_size(self, len: usize) -> ExactSize<Self> {
```

Comment:
The documentation should metion that the caller is responsible for the correctness of `len`, there is no validation whatsoever that the `len` is correct.

Reply:
Done. Also added a check against `size_hint` to at least have _some_ validation.